### PR TITLE
Ignition rendering poc

### PR DESCRIPTION
**Pull request by corest**
Towards: https://github.com/giantswarm/giantnetes-terraform/issues/130

We have now separate ignition files for aws/azure bastion/vault/master/worker nodes. So when we are changing something, it usually goes into both files. This creates some overhead in management as a lot of data is just duplicated. 

This is an attempt to solve this. The idea is to generate actual ignition files (which are rendered with vaules by terraform) via go templates. For this approach small utility is used (https://github.com/giantswarm/ignition-renderer/blob/renderer/main.go). Whenever template is changed - it is re-rendered for the proper platform (aws/azure) and then passed to the terraform. 

This PR is poc for managing bastion hosts. And it removes ~150 lines of duplicated ignition with templating.

This also has its disadvantages. With the single template, changes for one platform will cause terraform reapply for another platform. 

So, the question is do we really want this templating?

### CODE REVIEW

**Comment by calvix** [comment]
I like this, this makes sense to me, but maybe we can have the rendering tool as a terraform plugin which would simplify installing ( as terraform will do it for us). We do it similarly with ct plugin https://github.com/coreos/terraform-provider-ct

from my side LGTM if it works properly.

**Comment by corest** [comment]
@calvix I've looked into terraform plugins. It is much more complex and would be worth to do it in case of complex configs. In our case, we are using only one parameter(aws/azure) and that seems to be too simple for plugin. 

Regarding simplifying installing - with binary in the repository - there is no need to install anything additional.

**Comment by corest** [comment]
@r7vme 
Tried the plugin you've mentioned and I think we are on a good way with removing all the hacks

**Comment by corest** [comment]
Deployed to gauss and now deploying to azure. Seems to be good. There is only step for generating ignition from template and then for converting to cloudconfig.

ptal pls

File: templates/worker.yaml.tmpl [hunk 1]
```diff
@@ -487,9 +493,14 @@ systemd:
       --containerized \
       --enable-server \
       --logtostderr=true \
+      {{if eq .Provider "azure" -}}
+      --cloud-provider=azure \
+      --cloud-config=/etc/kubernetes/config/azure.yaml \
+      {{ else -}}
       --cloud-provider=aws \
+      --pod-infra-container-image={{ .PodInfraImage }} \
```
Comment (calvix) [inline templates/worker.yaml.tmpl]: isn't this more like general things rather than aws specific?

File: .circleci/config.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
 
   e2eTestAWS: &e2eTestAWS
     docker:
-      - image: quay.io/giantswarm/docker-terraform-and-stuff:latest
+      - image: quay.io/giantswarm/docker-terraform-and-stuff:gotemplate
```
Comment (calvix) [inline .circleci/config.yml]: You probably want to switch back to latest before the merge.

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ jobs:
 
   e2eTestAzure: &e2eTestAzure
     docker:
-      - image: quay.io/giantswarm/docker-terraform-and-stuff:latest
+      - image: quay.io/giantswarm/docker-terraform-and-stuff:gotemplate
     steps:
     - checkout
     - run: ./misc/e2e-azure.sh
```
Comment (calvix) [inline .circleci/config.yml]: You probably want to switch back to latest before the merge.

File: .circleci/config.yml [hunk 4]
```diff
@@ -36,7 +36,7 @@ jobs:
 
   e2eTestAzure: &e2eTestAzure
     docker:
-      - image: quay.io/giantswarm/docker-terraform-and-stuff:latest
+      - image: quay.io/giantswarm/docker-terraform-and-stuff:gotemplate
```
Comment (calvix) [inline .circleci/config.yml]: You probably want to switch back to latest before the merge.

**Comment by calvix** [comment]
looks much better than before, LGTM IFWY

File: .circleci/config.yml (hunk 2 above)
Comment (corest) [reply .circleci/config.yml]: sure

File: templates/worker.yaml.tmpl (hunk 1 above)
Comment (corest) [reply templates/worker.yaml.tmpl]: Yes, it is general. But used only on AWS as we are overriding default for China

**Pull request merged**